app/util/rpc-domain-utils.test.ts

```diff
@@ -1,152 +1,377 @@
+        mockNetworkController.getNetworkConfigurationByNetworkClientId.mockReturnValue({
+          rpcUrl: 'https://legacy-rpc.com',
+        });
+        (Engine.context as unknown as { NetworkController: MockNetworkController }).NetworkController = mockNetworkController;
+        // Exercise
+        const result = getNetworkRpcUrl('0x1');
+        // Verify
+        expect(result).toBe('https://legacy-rpc.com');
+      });
+      it('returns RPC URL from rpcEndpoints array', () => {
+        // Setup
+        const { mockNetworkController } = setupTestEnvironment();
+        mockNetworkController.findNetworkClientIdByChainId.mockReturnValue('network1');
+        mockNetworkController.getNetworkConfigurationByNetworkClientId.mockReturnValue({
+          rpcEndpoints: [
+            { url: 'https://rpc1.com' },
+            { url: 'https://rpc2.com' },
+          ],
+          defaultRpcEndpointIndex: 1,
+        });
+        (Engine.context as unknown as { NetworkController: MockNetworkController }).NetworkController = mockNetworkController;
+        // Exercise
+        const result = getNetworkRpcUrl('0x1');
+        // Verify
+        expect(result).toBe('https://rpc2.com');
+      });
+      it('returns unknown when network client ID not found', () => {
+        // Setup
+        const { mockNetworkController } = setupTestEnvironment();
+        mockNetworkController.findNetworkClientIdByChainId.mockReturnValue(null);
+        (Engine.context as unknown as { NetworkController: MockNetworkController }).NetworkController = mockNetworkController;
+        // Exercise
+        const result = getNetworkRpcUrl('0x1');
+        // Verify
+        expect(result).toBe('unknown');
+      });
+      it('returns unknown when network configuration not found', () => {
+        // Setup
+        const { mockNetworkController } = setupTestEnvironment();
+        mockNetworkController.findNetworkClientIdByChainId.mockReturnValue('network1');
+        mockNetworkController.getNetworkConfigurationByNetworkClientId.mockReturnValue(null);
+        (Engine.context as unknown as { NetworkController: MockNetworkController }).NetworkController = mockNetworkController;
+        // Exercise
+        const result = getNetworkRpcUrl('0x1');
+        // Verify
+        expect(result).toBe('unknown');
+      });
+      it('handles errors gracefully', () => {
+        // Setup
+        const { mockNetworkController } = setupTestEnvironment();
+        mockNetworkController.findNetworkClientIdByChainId.mockImplementation(() => {
+          throw new Error('Test error');
+        });
+        (Engine.context as unknown as { NetworkController: MockNetworkController }).NetworkController = mockNetworkController;
+        // Exercise
+        const result = getNetworkRpcUrl('0x1');
+        // Verify
+        expect(result).toBe('unknown');
+      });
+      it('handles missing rpcEndpoints gracefully', () => {
```

Comment:
S: same here about handle and gracefully
```suggestion
      it('returns unknown as RPC URL on missing rpcEndpoints', () => {
```